go-controller/pkg/ovn/egressip.go

```diff
@@ -299,7 +298,7 @@ func (oc *Controller) assignEgressIPs(eIP *egressipv1.EgressIP) error {
 	}()
 	assignableNodes, existingAllocations := oc.getSortedEgressData()
 	if len(assignableNodes) == 0 {
-		oc.eIPC.assignmentRetry.Store(eIP.Name, true)
+		oc.eIPC.assignmentRetry[eIP.Name] = true
```

Comment:
would it make more sense to lock here for the map instead?
oc.eIPC.assignmentRetryMutex.Lock()

then you can defer unlock in here as well.

Reply:
No, it will cause deadlocks with `addEgressNode`. Whenever a user labels a new node for egress assignment, we need to go over the `assignmentRetry` map and try to assign everything that has been unassigned until then, this is done in `addEgressNode`...however: since that ends up calling `assignEgressIPs` we will cause a deadlock if I try to lock `assignmentRetry` there too. 

This is all not very good, but I told myself I'd start refactoring this and cleaning it up once we move our code base to workqueues with a proper reconiliation logic.

Reply:
eek OK seems like we need to clean this up, but I'll trust for now this is correct